docs/devel/releasing.md

```diff
@@ -1,7 +1,155 @@
 # Releasing Kubernetes
 
-This document explains how to create a Kubernetes release (as in version) and
-how the version information gets embedded into the built binaries.
+This document explains how to cut a release, and the theory behind it. If you
+just want to cut a release and move on with your life, you can stop reading
+after the first section.
+
+## How to cut a Kubernetes release
+
+Regardless of whether you are cutting a major or minor version, cutting a
+release breaks down into four pieces:
+
+1. Selecting release components.
+1. Tagging and merging the release in Git.
+1. Building and pushing the binaries.
+1. Writing release notes.
+
+You should progress in this strict order without interleaving pieces of each
+major step unless you are very familiar with the release process.
+
+### Building a New Major/Minor Version (`vx.y.0`)
```

Comment:
Fixed. Feel free to drop in the minor nits, if you want. Though a large goal of this document is "something is better than nothing." And really, "something is better than 2 things", because #1883 and `releasing.md` were both acting as semi-canonical sources.